File: sockets/python/UserInterface.py
from time import sleep

class UI:

    def __init__(self):
        self.options = [
            "Genre",
            "Lead studio",
            "Audience score %",
            "Profitability",
            "Rotten tomatoes %",
            "Worldwide gross",
            "Year"
        ]

    def mainMenu(self):
        invalidOption = True
        selectedOption = None

        while invalidOption:
            print("\t*** Sockets in python ***\n")
            print("Please select an option.")
            print("[1] Make a query about a movie")
            print("[2] exit")
            try:
                selectedOption = int(input("Option: "))
                print("")
            except:
                print("Invalid option selected.")
                sleep(2)
                continue

            if selectedOption > 2 or selectedOption < 1:
                print("Invalid option selected.")
                sleep(2)
                continue

            invalidOption = False

        return selectedOption

    def getMoviesName(self):

        selectedMovie = input(
            f'Please specify the movie you want to make a query about: ')

        return selectedMovie

    def getMovieOption(self):

        invalidOption = True
        selectedOption = None

        while invalidOption:
            print("Please specify .")
            for i in range(1, 7):
                print(f"[{i}] {self.options[i - 1]}")
            try:
                selectedOption = int(input("Option: "))
                print("")
            except:
                print("Invalid option selected.")
                sleep(2)
                continue

            if selectedOption > 7 or selectedOption < 1:
                print("Invalid option selected.")
                sleep(2)
                continue

            invalidOption = False

        return selectedOption

File: sockets/ruby/UserInterface.rb
class UserInterface
  def initialize
    @options = [
      "Genre",
      "Lead studio",
      "Audience score %",
      "Profitability",
      "Rotten tomatoes %",
      "Worldwide gross",
      "Year",
    ]
    @movie  = ""
    @option = 0
  end

  def mainMenu
    invalidOption  = true
    selectedOption = nil

    while invalidOption
      puts "\t*** Sockets in ruby ***\n"
      puts "Please select an option."
      puts "[1] Make a query about a movie"
      puts "[2] exit"
      puts "Option: "
      selectedOption = gets

      selectedOption = selectedOption.to_i

      if selectedOption < 1 or selectedOption > 2
        puts "Invalid option selected."
        sleep 2
        next
      end
      invalidOption = false
    end

    selectedOption
  end

  def getMoviesName
    puts "Please specify the movie you want to make a query about: "
    @movie = gets
    @movie
  end

  def getMovieOption
    invalidOption = true
    selectedOption = nil

    while invalidOption
    
      puts "Please specify the attribute of the movie you need."
      i = 1
      @options.each do |option| 
        puts "[#{i}] #{option}"
        i += 1
     end
      puts "Option: "
      selectedOption = gets

      selectedOption = selectedOption.to_i

      if selectedOption < 1 or selectedOption > 7
        puts "Invalid option selected."
        sleep 2
        next
      end
      invalidOption = false
    end
    @option = selectedOption - 1
    selectedOption
  end

  def displayResponse(res)
    splitRes = res.split("|")

    success = splitRes.first.to_i == 0 ? false : true

    if success
      puts "The #{@options[@option]} for the movie #{@movie.chomp} is: #{splitRes.last}\n\n"
    else
      puts "The #{@options[@option]} for the movie #{@movie.chomp} was not found.\n\n"
    end

  end
end
